Parsers/ESETPROTECT/ESETPROTECT.txt

```diff
@@ -0,0 +1,159 @@
+//      <10>Mar 17 09:42:53 ESMC-linux ERAServer[12412]: {"event_type":"Threat_Event","ipv4":"10.0.0.6","hostname":"azwin10vm.lab.local","source_uuid":"763a89ef-f8dd-41a4-9b8c-0e3bf34437d5","occured":"17-Mar-2021 09:42:45","severity":"Critical","threat_type":"test file","threat_name":"Eicar","scanner_id":"Real-time file system protection","scan_id":"virlog.dat","engine_version":"22977P (20210317)","object_type":"file","object_uri":"file:///C:/Users/donny/Desktop/eicar.com","action_taken":"retained","threat_handled":false,"need_restart":false,"username":"lab\\donny","processname":"C:\\Windows\\explorer.exe","circumstances":"Event occurred on a newly created file.","firstseen":"06-Jan-2021 09:06:28","hash":"3395856CE81F2B7382DEE72602F798B642F14140"}
+//      <14>Mar 17 09:46:49 ESMC-linux ERAServer[12412]: {"event_type":"Audit_Event","ipv4":"10.0.0.4","hostname":"ESMC-linux","source_uuid":"c216d4e0-3469-4916-a469-923f4bd1cf9b","occured":"17-Mar-2021 09:46:49","severity":"Information","domain":"Domain group","action":"Login attempt","target":"82764bdf-f5cf-477b-be47-6a670bcbf30e","detail":"Authenticating domain user 'donny'.","user":"","result":"Success"}
+//
+
+//
+// Get event data
+//
+let LogHeader = Syslog
+    | where ProcessName == "ERAServer"
+    | project-rename EventProduct = ProcessName 
+    | extend EventVendor = "ESET",
+        Message = parse_json(SyslogMessage)
+    | extend EventType = Message['event_type'],
+        DvcIpAddr = Message['ipv4'],
+        EventResourceId = Message['source_uuid'],
+        EventCreationTime = make_datetime(Message['occured']),
+        EventSeverity = Message['severity'],
+        DvcHostname = Message['hostname'];
+//
+// Threat events
+//
+let ThreatEvents = LogHeader
+    | where EventType == "Threat_Event"
+    | extend ThreatCategory = Message['threat_type'],
+        ThreatName = Message['threat_name'],
+        ThreatFlags = Message['threat_flags'],
+        EventSubType = Message['scanner_id'],
+        EngineVersion = Message['engine_version'],
+        ObjectType = Message['object_type'],
+        FilePath = Message['object_uri'],
+        EventResult = Message['action_taken'],
+        EventError = Message['action_error'],
+        ScanId = Message['scanner_id'],
+        ThreatHandled = Message['threat_handled'],
+        NeedRestart = Message['need_restart'],
+        ProcessFilePath = Message['processname'],
+        EventMessage = Message['circumstances'],
+        FileHashSha1 = Message['hash'],
+        UserDomain = iff(Message['username'] contains '\\', split(Message['username'], '\\')[0], ''),
```

Comment:
It should be either DstUserDomain or SrcUserDomain.

Thanks,
Aditi Maheshwari